docs/arch/01-deployment-modes.md

```diff
@@ -0,0 +1,479 @@
+    style STS fill:#ffb74d
+```
+
+### How It Works
+
+1. **User applies CRD**: `kubectl apply -f mcpserver.yaml`
+
+2. **Operator watches resources** (`cmd/thv-operator/controllers/mcpserver_controller.go`):
+   - Watches for `MCPServer` custom resources
+   - Reconciles desired state vs actual state
+
+3. **Operator creates Deployment**:
+   - Runs `thv-proxyrunner` container
+   - Mounts RunConfig as ConfigMap or secret
+   - Applies middleware configuration
+
+4. **Proxy runner creates StatefulSet**:
+   - Uses Kubernetes API (in-cluster client)
+   - Creates StatefulSet with MCP server container
+   - Manages container lifecycle
+
+5. **Proxy runner proxies traffic**:
+   - Receives requests on exposed port
+   - Applies middleware chain
+   - Forwards to StatefulSet pod(s)
+
+6. **Operator creates Service**:
+   - Exposes proxy runner Deployment
+   - LoadBalancer, ClusterIP, or NodePort
+   - Routes external traffic to proxy
+
+### Why Two Binaries?
+
+**`thv-operator`** (`cmd/thv-operator/`):
+- Watches Kubernetes API for CRDs
+- Reconciles desired vs actual state
+- Creates Kubernetes resources (Deployments, Services, ConfigMaps)
+- Does NOT run the proxy or create containers directly
+
+**`thv-proxyrunner`** (`cmd/thv-proxyrunner/`):
+- Runs as a container in the Deployment
+- Creates containers via Kubernetes API (StatefulSets)
+- Applies middleware and proxies MCP traffic
+- Handles transport-specific communication
+
+**Why not use `thv` in Kubernetes?**
+- `thv` is optimized for local Docker/Podman API usage
+- Kubernetes requires different container creation logic (StatefulSets vs standalone containers)
+- Separation of concerns: operator manages K8s resources, proxy-runner manages MCP traffic
+
+### Deployment Pattern
+
+```mermaid
+graph LR
+    subgraph "Namespace: default"
+        Deploy[Deployment<br/>proxy-runner<br/>Replicas: 1]
+        SVC[Service<br/>proxy-svc]
+        STS[StatefulSet<br/>mcp-server<br/>Replicas: 1]
+    end
+
+    Deploy -->|manages| STS
+    SVC -->|routes to| Deploy
+    Deploy -.->|watches| STS
+
+    style Deploy fill:#90caf9
+    style STS fill:#ffb74d
+    style SVC fill:#81c784
+```
+
+### Custom Resource Definitions
+
+ToolHive provides several CRDs for managing MCP servers in Kubernetes:
+
+- **MCPServer** - Defines an MCP server deployment with container images, transports, and middleware
+- **MCPRegistry** - Manages MCP server registries from Git or ConfigMap sources
+
+For complete examples, see the [`examples/operator/mcp-servers/`](../../examples/operator/mcp-servers/) directory, which includes:
+- Basic MCP server deployments with different transports (stdio, SSE, streamable-http)
+- Authentication configurations (inline OIDC, ConfigMap-based, Kubernetes-native)
+- Resource and pod template customizations
+- Tool filtering and middleware examples
+
+Full CRD API documentation is available in [`cmd/thv-operator/crd-api.md`](../../cmd/thv-operator/crd-api.md).
```

Comment:
The referenced file `cmd/thv-operator/crd-api.md` does not exist. Consider updating to the correct path.

```suggestion
Full CRD API documentation is available in `docs/operator/crd-api.md`.
```

Evidence: File `cmd/thv-operator/crd-api.md` does not exist in the repository.